packages/builder/src/pages/builder/workspace/[application]/agent/[agentId]/config.svelte

```diff
@@ -92,43 +86,59 @@
   }
 
   async function toggleAgentLive() {
-    if (!currentAgent) return
+    if (!currentAgent || togglingLive) return
+
+    const nextLive = !currentAgent.live
 
     try {
-      if (currentAgent.live) {
-        await agentsStore.updateAgent({
-          ...currentAgent,
-          ...draft,
-          live: false,
-        })
-        notifications.success("Agent has been paused")
-      } else {
-        await agentsStore.updateAgent({
-          ...currentAgent,
-          ...draft,
-          live: true,
-        })
-        notifications.success("Agent is now live")
-      }
+      togglingLive = true
+
+      await agentsStore.updateAgent({
+        ...currentAgent,
+        ...draft,
+        live: nextLive,
+      })
+      await deploymentStore.publishApp()
       await agentsStore.fetchAgents()
+
+      notifications.success(
+        nextLive ? "Agent is now live" : "Agent has been paused"
+      )
     } catch (error) {
       console.error(error)
       notifications.error(
-        currentAgent.live ? "Error pausing agent" : "Error setting agent live"
+        nextLive ? "Error setting agent live" : "Error pausing agent"
       )
+    } finally {
+      togglingLive = false
     }
   }
 
-  const confirmDeleteToolSource = (e: PointerEvent, toolSource: any) => {
-    e.stopPropagation()
-    toolSourceToDelete = toolSource
-    deleteConfirmModal.show()
+  function getToolSourceIcon(toolSource: AgentToolSource) {
+    if (toolSource.type === ToolSourceType.REST_QUERY) {
+      const ds = $datasources.list.find(d => d._id === toolSource.datasourceId)
+      if (ds?.restTemplate) {
+        return getIcon(
+          IntegrationTypes.REST,
+          ds.restTemplate,
+          restTemplates.getByName(ds.restTemplate)?.icon
+        )
+      }
+    }
+    return { icon: Budibase }
   }
 
+  const confirmDeleteToolSource =
+    (toolSource: AgentToolSource) => (e: MouseEvent) => {
+      e.stopPropagation()
+      toolSourceToDelete = toolSource
+      deleteConfirmModal.show()
+    }
+
   const deleteToolSource = async () => {
     if (!toolSourceToDelete) return
     try {
-      await agentsStore.deleteToolSource(toolSourceToDelete.id)
+      await agentsStore.deleteToolSource(toolSourceToDelete.id!)
```

Comment:
NAB: Maybe in the line above do `if (!toolSourceToDelete?.id) return` to avoid the bang here?